app/assets/javascripts/discourse/app/lib/sidebar/base-custom-sidebar-button.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Base class representing a sidebar section button interface.
+ */
+export default class BaseCustomSidebarSectionButton {
+  constructor({ sidebar, router } = {}) {
+    this.sidebar = sidebar;
+    this.router = router;
+  }
+
+  /**
+   * @returns {string} The label of the section button
+   */
+  get label() {
+    this._notImplemented();
+  }
+
+  /**
+   * @returns {string} The icon of the section button
+   */
+  get icon() {
+    this._notImplemented();
+  }
+
+  /**
+   * @returns {Function} Action for button
+   */
+  get action() {
+    this._notImplemented();
+  }
+
+  _notImplemented() {
```

Comment:
```suggestion
  #notImplemented() {
```

Native JS class has private functions :)